sklearn/decomposition/tests/test_incremental_pca.py

```diff
@@ -74,40 +74,41 @@ def test_incremental_pca_inverse():
 
 def test_incremental_pca_validation():
     # Test that n_components is >=1 and <= n_features.
-    X = [[0, 1, 0], [1, 0, 0]]
+    X = np.array([[0, 1, 0], [1, 0, 0]])
+    n_samples, n_features = X.shape
     for n_components in [-1, 0, .99, 4]:
         assert_raises_regex(ValueError,
-                            "n_components\=.* invalid for n_features\=.*, need"
+                            "n_components={} invalid for n_features={}, need"
                             " more rows than columns for IncrementalPCA "
-                            "processing",
+                            "processing".format(n_components, n_features),
                             IncrementalPCA(n_components, batch_size=10).fit, X)
 
     # Tests that n_components is also <= n_samples.
+    n_components = 3
     assert_raises_regex(ValueError,
-                        "n_components\=.* be less or equal to "
-                        "the batch number of samples .*\.",
-                        IncrementalPCA(n_components=3).partial_fit, X)
+                        "n_components={} must be less or equal to "
+                        "the batch number of samples {}".format(
+                            n_components, n_samples),
+                        IncrementalPCA(
+                            n_components=n_components).partial_fit, X)
 
 
 def test_n_components_none():
     # Ensures that n_components == None is handled correctly
     rng = np.random.RandomState(1999)
     for n_samples, n_features in [(50, 10), (10, 50)]:
-
+        X = rng.rand(n_samples, n_features)
```

Comment:
Thanks for all the improvements. For this bit here, the reason why I wanted X to be created anew for each partial_fit call was just to replicate the mechanism of batches (i.e. if you get three batches of 30 samples from a dataset with 90 samples, the three batches will be different).

Not sure it's that important for our case, but regardless I just realised my code was producing identical batches anyway, since the random state is fixed beforehand (I guess I would have had to use a different random state for the second call)